earth_enterprise/src/common/notify.cpp

```diff
@@ -68,6 +69,55 @@ khNotifyLevel getNotifyLevel() {
   return notifyLevel;
 }
 
+std::string khNotifyLevelToString(khNotifyLevel level)
+{
+    std::string retval;
+    switch(level)
+    {
+    case NFY_FATAL: retval = "NFY_FATAL"; break;
+    case NFY_WARN: retval = "NFY_WARN"; break;
+    case NFY_PROGRESS: retval = "NFY_PROGRESS"; break;
+    case NFY_INFO: retval = "NFY_INFO"; break;
+    case NFY_INFO2: retval = "NFY_INFO2"; break;
+    case NFY_DEBUG: retval = "NFY_DEBUG"; break;
+    case NFY_VERBOSE: retval = "NFY_VERBOSE"; break;
+    case NFY_NOTICE:
+    default: retval = "NFY_NOTICE";
+    };
+    return retval;
+}
+
+khNotifyLevel stringTokhNotifyLevel(const std::string& _level)
+{
+    khNotifyLevel retval;
+    std::locale loc;
+    std::string level(_level);
+
+    // make it case-insensitive
+    for(std::string::size_type i = 0; i < level.length(); ++i)
+        level[i] = std::toupper(level[i],loc);
+
+    if (level == "NFY_FATAL")
+        retval = NFY_FATAL;
+    else if (level == "NFY_WARN")
+        retval = NFY_WARN;
+    else if (level == "NFY_PROGRESS")
+        retval = NFY_PROGRESS;
+    else if (level == "NFY_INFO")
+        retval = NFY_INFO;
+    else if (level == "NFY_INFO2")
+        retval = NFY_INFO2;
+    else if (level == "NFY_DEBUG")
+        retval = NFY_DEBUG;
+    else if (level == "NFY_VERBOSE")
+        retval = NFY_VERBOSE;
+    else if (level == "NFY_NOTICE")
+        retval = NFY_NOTICE;
+    else if (level == "NFY_DEFAULT_LEVEL")
```

Comment:
I'm getting a compile error on CentOS 6 here because retval may be uninitialized. If you remove the "if" part of this "if else" statement (so that it returns NFY_DEFAULT_LEVEL for any string that doesn't match one of the others), that fixes the build error.